From 0af7c1340b6a659b1525d444306725671162ba4f Mon Sep 17 00:00:00 2001
From: Aichun Shi <aichun.shi@intel.com>
Date: Thu, 20 Oct 2022 16:39:21 +0800
Subject: [PATCH 0313/1646] [Backport] vfio/type1: Fix kabi issue for
 vfio_iommu_driver_ops

Intel inclusion
category: feature
bugzilla: https://gitee.com/openeuler/intel-kernel/issues/I5BEBU

Reference: https://gitee.com/openeuler/kernel/commit/d9c53b016b66e21d094d9a2e3a51391129e4f943

-------------------------------------

Use KABI_EXTEND to fix kabi issue for struct vfio_iommu_driver_ops.

Signed-off-by: Aichun Shi <aichun.shi@intel.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
---
 include/linux/vfio.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/vfio.h b/include/linux/vfio.h
index 29a11f2179d0..e57ffde04f5d 100644
--- a/include/linux/vfio.h
+++ b/include/linux/vfio.h
@@ -106,8 +106,8 @@ struct vfio_iommu_driver_ops {
 					       struct notifier_block *nb);
 	int		(*dma_rw)(void *iommu_data, dma_addr_t user_iova,
 				  void *data, size_t count, bool write);
-	struct iommu_domain *(*group_iommu_domain)(void *iommu_data,
-						   struct iommu_group *group);
+	KABI_EXTEND(struct iommu_domain *(*group_iommu_domain)(void *iommu_data,
+							       struct iommu_group *group))
 };
 
 extern int vfio_register_iommu_driver(const struct vfio_iommu_driver_ops *ops);
-- 
2.17.1

